lib/Widgets/SettingsPage/AbstractSimpleSettingsPage.vala

```diff
@@ -0,0 +1,159 @@
+    private Gtk.Image header_icon;
+    private Gtk.Label description_label;
+    private Gtk.Label title_label;
+    private string _description;
+    private string _icon_name;
+    private string _title;
+
+    /**
+     * A #Gtk.ButtonBox used as the action area for #this
+     */
+    public Gtk.ButtonBox action_area { get; construct; }
+
+    /**
+     * A #Gtk.Grid used as the content area for #this
+     */
+    public Gtk.Grid content_area { get; construct; }
+
+    /**
+     * A #Gtk.Switch that appears in the header area when #this.activable is #true
+     */
+    public Gtk.Switch status_switch;
+
+    /**
+     * Creates a #Gtk.Switch #status_switch in the header of #this
+     */
+    public bool activatable { get; construct; }
+
+    /**
+     * Creates a #Gtk.Label with a page description in the header of #this
+     */
+    public string description {
+        get {
+            return _description;
+        } 
+        construct set {
+            if (description_label != null) {
+                description_label.label = value;
+            }
+            _description = value;
+        }
+    }
+
+    /**
+     * An icon name associated with #this
+     */
+    public string icon_name {
+        get {
+            return _icon_name;
+        } 
+        construct set {
+            if (header_icon != null) {
+                header_icon.icon_name = value;
+            }
+            _icon_name = value;
+        }
+    }
+
+    /**
+     * A title associated with #this
+     */
+    public string title {
+        get {
+            return _title;
+        } 
+        construct set {
+            if (title_label != null) {
+                title_label.label = value;
+            }
+            _title = value;
+        }
+    }
+
+    /**
+     * Creates a new SimpleSettingsPage
+     */
+    public SimpleSettingsPage () {
+        Object (activatable: activatable,
+                icon_name: icon_name,
+                description: description,
+                title: title);
+    }
+
+    construct {
+        header_icon = new Gtk.Image.from_icon_name (icon_name, Gtk.IconSize.DIALOG);
+        header_icon.pixel_size = 48;
+        header_icon.valign = Gtk.Align.START;
+
+        title_label = new Gtk.Label (title);
+        title_label.xalign = 0;
+        title_label.get_style_context ().add_class ("h2");
+
+        var header_area = new Gtk.Grid ();
+        header_area.column_spacing = 12;
+        header_area.row_spacing = 3;
+        header_area.attach (header_icon, 0, 0, 1, 2);
+        header_area.attach (title_label, 1, 0, 1, 1);
+
+        if (description != null) {
+            description_label = new Gtk.Label (description);
+            description_label.xalign = 0;
+            description_label.wrap = true;
+
+            header_area.attach (description_label, 1, 1, 1, 1);
+        }
+
+        if (activatable) {
+            status_switch = new Gtk.Switch ();
```

Comment:
Okay, then. So you should make `status_switch` nullable, and point out in the documentation that it will be `null` when `activatable` is set to `false`